client/my-sites/stats/stats-date-picker/date-label-drill/index.tsx

```diff
@@ -0,0 +1,43 @@
+import page from '@automattic/calypso-router';
+import { Icon, levelUp } from '@wordpress/icons';
+import { useState, useEffect, ReactNode } from 'react';
+import { appendQueryStringForRedirection } from '../../utils';
+import './index.scss';
+
+interface QueryParams {
+	drilldown?: string;
+}
+
+interface DateLabelDrillProps {
+	query?: QueryParams;
+	children?: ReactNode;
+}
+
+const DateLabelDrill = ( { query, children }: DateLabelDrillProps ) => {
+	const [ isAnimated, setIsAnimated ] = useState( false );
+
+	useEffect( () => {
+		setIsAnimated( true );
+
+		const params = { ...query };
+		delete params.drilldown;
+		const url = appendQueryStringForRedirection( window.location.pathname, params );
```

Comment:
We should probably use `context.path` here?

Reply:
I was considering that was an option to pass `context` as the prop. 👍🏼

Reply:
Updated in [5d2e369](https://github.com/Automattic/wp-calypso/pull/97906/commits/5d2e36936e570ad6767a5327f055254417099f48).

Reply:
Finally, I got back to using `sessionStorage` and `useMemo` to resolve the re-rendering issue I was stuck in: [26a79ee](https://github.com/Automattic/wp-calypso/pull/97906/commits/26a79ee8caa75df64910cce68081a73604a8f832).